Chinese_Chessv1.0

zyb-0609 opened the pull request:
# ♟️ LVGL Chinese Chess App

基于 [LVGL](https://lvgl.io/) 图形库实现的中国象棋应用程序。  
本项目支持棋盘绘制、棋子显示、点击交互、规则判断、以及游戏结束提示窗口等功能。  
适用于嵌入式系统、模拟器或 PC 端 LVGL 环境。

---

## 🧩 功能概述

- 🎨 **图形界面**
  - 基于 LVGL 绘制棋盘、棋子和状态窗口
  - 棋盘背景、棋子图片资源独立管理
  - 游戏结束时显示弹窗提示

- ♟️ **象棋逻辑**
  - 棋子移动规则判断（车、马、象、士、将、炮、兵）
  - 阵营切换控制（红方 / 黑方）
  - 棋子吃子与死亡状态更新
  - 棋盘状态检查与“将/帅”生存判断

- 🖱️ **交互控制**
  - 点击棋子与移动判定
  - 支持重新开始和退出事件

---

jianglianfang committed:
move bandx and music_player

Change-Id: I63b520d862876b382cdc5b891dead37b6293b2cd

jianglianfang committed:
bandx: fix music page progress arc issue

Change-Id: Iebd904530afc6975536843c7b8ed8a4880a423cb

jianglianfang committed:
bandx: use new assets

Change-Id: Ic2d791df06c553ec801965aa41d0ed8177f2a411

jianglianfang committed:
music_player:fix warning

Change-Id: I8bf1111e3b8c2ffb2577f0f53fc449198f023757

FASTSHIFT committed:
packages: porting X-TRACK project

Change-Id: Idbdddf27f976377265ad215b9a0f7eefa7b4a8e4

openvela-robot committed:
add ci.yml

Change-Id: Iaf1416ae9e9b4c06ab290ff4c17d1f6585fc3c29

FASTSHIFT committed:
packages: add LVGL duplicate initialization protection

Change-Id: I1decaf8a2d2f017f44bfb25b9300e2ac977db3f0
Signed-off-by: pengyiqiang <pengyiqiang@xiaomi.com>

W-Mai committed:
music_player: add resource healthy check

Change-Id: I0d12e042a6897edc15122e110fdcce34172d3615
Signed-off-by: xinbingnan <xinbingnan@xiaomi.com>

openvela-robot committed:
add issue template

openvela-robot committed:
update workflows ci

openvela-robot committed:
update issue template

openvela-robot committed:
workflows add checkpatch.yaml

liujinye-sys committed:
workflows add ci.yaml

HongChao6 committed:
bandx: free bandx all resources.

Signed-off-by: HongChao6 <liuhongchao_00@126.com>

HongChao6 committed:
music_player: use deinit free all resources.

Signed-off-by: HongChao6 <liuhongchao_00@126.com>

HongChao6 committed:
x_track: use nuttx_deinit to free all resources for x_track.

Signed-off-by: HongChao6 <liuhongchao_00@126.com>

tanghao-xiaomi committed:
Add Readme file

tanghao-xiaomi committed:
Add Readme file

tanghao-xiaomi committed:
Add Readme file

tanghao-xiaomi committed:
Add Readme file

tanghao-xiaomi committed:
Add Readme file

openvela-robot committed:
update ci pull_request types to [opened, reopened, synchronize]

openvela-robot committed:
add .github/CODEOWNERS PULL_REQUEST_TEMPLATE.md

openvela-robot committed:
update workflows ci.yml, delete pull_request_review trigger

openvela-robot committed:
Update the ci trigger from pull_request to pull_request_target

foamzzz committed:
calculator-foamzzz

foamzzz committed:
calculator_ExpressionCalc-foamzzz

foamzzz committed:
no longer use global variables

foamzzz committed:
Fix memory leak, add image and font support, enhance UI quality.

foamzzz committed:
using C++ and made changes to the UI.

jiayadong8995 committed:
create relation calculator

Hu-JW committed:
Whackmole

调整hammer和mole的大小

Updated game over message box and font style

Supplementary English notes

Supplementary notes in English

Supplementary notes in English for makefile

delete stopwatch

update mole picture

update UI

forget

add whackmole.gif , readme and Code comments.

delete some code comments

update makefile and make.defs

update whackmole.gif

update readme.md

update menuconfig

format codes

delete unused codes

format codes

Update Readme.md

Update Whackmole_main.c

openvela-robot committed:
workflows add stale.yml

foamzzz committed:
Support for negative numbers, disabling try-catch, and optimizing input after calculation.

foamzzz committed:
Fix the delay issue after adding PNG images. PI and E can be used directly.

foamzzz committed:
Remove warning

foamzzz committed:
Enhanced Del key functionality. Switched to pow for exponentiation.

foamzzz committed:
Update Readme.md

Update the description to match the merged version.

liujinye-sys committed:
CODEOWNERS add jianglianfang

jiayadong8995 committed:
fix unexpected result

modify comments

modify comments

zhushiyi23 committed:
Update README.md

zhushiyi23 committed:
Update README_zh-cn.md

zhushiyi23 committed:
Update README.md

zhushiyi23 committed:
Update README_zh-cn.md

7ZhangZY committed:
feat: complete electronic-pet demo with states and animations

Signed-off-by: ZZY <18997558607@163.com>

xk19 committed:
Snake game implementation and optimizations
﻿
* Initial snake game implementation
* Optimized game performance
* Updated README with instructions

chenzepei517 committed:
fix not found page

jiayadong8995 committed:
apate esp32s3-box developemnt board and update readme

add .c font

apate esp32s3-box developemnt board

apate esp32s3-box developemnt board

TangMeng12 committed:
Update README.md and README_zh-cn.md

Hu-JW committed:
suceessfully running on esp

Signed-off-by: Hu_JW <1748239230@qq.com>

suceessfully on esp

successfully on esp

successfully running on esp

successfully running on esp

update readme

update readme

update readme

code format

jiayadong8995 committed:
fix readme file path

openvela-robot committed:
update .github

openvela-robot committed:
update .github

tanghao-xiaomi committed:
Whack-a-Mole Demo Quick Start

tanghao-xiaomi committed:
Relatives Calculator Demo Quick Start

TangMeng12 committed:
Whackmole: Fix syntax errors in the code

Signed-off-by: v-tangmeng <v-tangmeng@xiaomi.com>

foamzzz committed:
v3.2 Fix ball movement bug, add level-clearing item.

tanghao-xiaomi committed:
Optimize document content

yanxingyu17 committed:
wooden_fish

lvy010 committed:
feat: 添加音乐播放器2项目
- 添加了项目代码
- 更新README.md和README_zh-cn.md，在使用说明中添加了链接

yanxingyu17 committed:
fix: remove duplicate lv_freetype_init and fix programmer mode auto-tap

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/open-vela/packages_demos?pullRequest=58) <br/>All committers have signed the CLA.

lvy010 committed:
feat: Fix the compilation path issue and standardize the code format

lvy010 committed:
docs: Change the project name to music_player2 and synchronize the Chinese and English files.

lvy010 committed:
Updated the document to the corresponding version

W-Mai commented on Chinese_Chess/res/.DS_Store:
ignore this

zyb-0609 replied:
> ignore this

  Hello, thank you very much for your suggestion! I have already ignored the Chinese_Chess/res/.DS_Store file in my submitted Chinese_Chess_v3.3 version.

W-Mai commented:
Please format code with Webkit style

yanxingyu17 committed:
fix music_player2

l-0823 committed:
Chinese_Chess_v2

Chinese_ChessV3

Chinese_V3.1

Chinese_chess_V3.2

Chinese_Chess_v3.3

Chinese_Chess_V3.4

Chinese_Chess_v3.4

Chinese_Chess_v4.0

Chinese_Chess_v4.1

W-Mai commented:
Please update a sample video or gif to show what you have done on your pr or readme. 😁

W-Mai commented:
Hi, you need sign the CLA first, thank you 😁

https://github.com/open-vela/packages_demos/pull/58#issuecomment-3414692080

zyb-0609 commented:
> Please format code with Webkit style

Hello, thank you very much for your suggestion! In my submitted Chinese_Chess_v3.3 version, I have modified the code to follow the Webkit style.

zyb-0609 commented:
> Please update a sample video or gif to show what you have done on your pr or readme. 😁

  Hello, thank you very much for your suggestion! In my submitted Chinese_Chess_v3.3 version, I have provided both Chinese and English versions of the README file as well as the version update file (i.e., the Update file). Additionally, I have added GIF animations to give a simple demonstration of the program’s features.

zyb-0609 commented:
> Please update a sample video or gif to show what you have done on your pr or readme. 😁

  Hello, thank you for the reminder. All accounts participating in the update have now signed the CLA agreement.

W-Mai commented:
Found a small problem, suggest showing who the winner is when the game ends.

zyb-0609 commented:
> Found a small problem, suggest showing who the winner is when the game ends.

  Hello, thank you for your suggestion! I have made the modifications in my latest commit — the program can now display the winning side when the game ends.

tanghao-xiaomi committed:
Convert document paths to relative

Commit:
openvela: 电子沙漏应用

jiayadong8995 commented on Chinese_Chess/Chinese_chess.h:
```diff
@@ -0,0 +1,172 @@
+extern int chess,chess2;
+extern int turn_camp; 
+extern pieceStruct piece[32];
+extern pieceStruct1 piece_struct[32];
```
They are only used in Chinese_chess.c, and you can prevent these variables from being exposed without using 'extern'.

jiayadong8995 commented on Chinese_Chess/Chinese_chess.h:
```diff
@@ -0,0 +1,172 @@
+bool init_resource(void);
+void app_create_main_page(void);
+void app_create_error_page(void);
+void app_create(void);
```
Chinese_chesss.h file only needs to expose one function interface:'app_create()'. It seems your other functions are called internally.
And please rename your interface to 'xxx_app_create()'.

jiayadong8995 commented:
please squash your change into one patch by git rebase -i HEAD~8

jiayadong8995 commented on Chinese_Chess/Chinese_chess.c:
```diff
@@ -0,0 +1,1353 @@
+struct ctx_s        C;  /**< Context */
+
+/*Instantiate chess pieces*/
+lv_obj_t* Red_car_1;
```
static

zyb-0609 replied:
> static

Hello, thank you for your suggestion. I have already added the static declaration when defining the variable.

jiayadong8995 commented on Chinese_Chess/Chinese_chess.c:
```diff
@@ -0,0 +1,1353 @@
+    lv_obj_del_async(win);  
+}
+
+
```
delete extra blank lines

zyb-0609 replied:
> delete extra blank lines

Hello, thank you for your suggestion. I have already removed the extra blank lines.

jiayadong8995 commented on Chinese_Chess/Chinese_chess.c:
```diff
@@ -0,0 +1,1353 @@
+    LV_STYLE_PROP_FLAG_NONE
+};
+
+
```
delete extra blank lines

l-0823 committed:
Chinese_Chess_v4.1

l-0823 committed:
Chinese_Chess_v4.1
  There are extra blank lines in the files Chinese_Chess.h and Chinese_Chess.c.
Add content
  Remove the blank lines in Chinese_Chess.h and Chinese_Chess.c.

TangMeng12 committed:
update the README file

1. Update the README file for the newly added demos.
2. Reorder demos bt string.

Signed-off-by: v-tangmeng <v-tangmeng@xiaomi.com>

Astra-dong committed:
modify:package/demos/pet

- modify readme to support openvela qemu
- fix compilation warning

Signed-off-by: Astra-dong <xbdieiong83@gmail.com>

Astra-dong committed:
modify wooden_fish and music_player

- modify wooden_fish picture
- add 'static' to avoid exposing variables.
- add utils

Signed-off-by: Astra-dong <xbdieiong83@gmail.com>

C-Luv9615 committed:
add calculator_use_builtin_res

jianglianfang commented:
It seems that v1 is **missing** from the commit message, only v2 and v4 are available

jianglianfang commented on Chinese_Chess/res/fonts/MiSans-Semibold.ttf:
where do you use these font resource?

zyb-0609 replied:
> where do you use these font resource?
The image resources I used were created by myself using basic shapes and are not subject to any usage restrictions. I have also updated the README to clarify the image sources.

jianglianfang commented on Chinese_Chess/Chinese_chess.h:
```diff
@@ -0,0 +1,100 @@
+
+};
+
+struct resource_s {   
```
Why do we need nested structures

zyb-0609 replied:
> Why do we need nested structures

Hello, the purpose of using this nested struct was to facilitate future resource extensions. Since this has not been implemented yet, I have revised the structure in the latest update of the code.

l-0823 committed:
Chinese_Chess_v4.2
Update README (CN/EN):
The chessboard and piece assets were all created by the author using basic
geometric shapes and are free of usage restrictions.

zyb-0609 commented:
> It seems that v1 is **missing** from the commit message, only v2 and v4 are available

Due to an incorrect Git command operation, version v1 can no longer be located. Versions v2 through v4.0 have been consolidated using the previously proposed git rebase -i HEAD~8 command.

l-0823 committed:
Chinese_Chess_v4.2

l-0823 committed:
Chinese_Chess_v4.2

l-0823 committed:
Chinese_Chess_v4.2
